proto/data_objects_service.proto

```diff
@@ -1,195 +1,274 @@
+  // checksum. If the checksum type is provided
+  ChecksumRequest checksum = 3;
+
+  // OPTIONAL
+  // Specifies the maximum number of results to return in a single page.
+  // If unspecified, a system default will be used.
+  int32 page_size = 4;
+
+  // OPTIONAL
+  // The continuation token, which is used to page through large result sets.
+  // To get the next page of results, set this parameter to the value of
+  // `next_page_token` from the previous response.
+  string page_token = 5;
+
 }
 
-message GetDataObjectCloudRequest {
-  string id = 1;
-  string cloud = 2;
+///-------------------------------------------------------------------
+// A list of data objects
+///-------------------------------------------------------------------
+message ListDataObjectsResponse {
+  // The list of Data Objects.
+  repeated DataObject data_objects = 1;
+
+  // The continuation token, which is used to page through large result sets.
+  // Provide this value in a subsequent request to return the next page of
+  // results. This field will be empty if there aren't any additional results.
+  string next_page_token = 2;
 }
 
-message DataObjectNativeURLResponse {
-  string url = 1;
-  string key = 2;
-  string secret_key = 3;
+// The Data Object one would like to index. One must provide any aliases
+// and URLs to this file when sending the CreateDataObjectRequest. It is up
+// to implementations to validate that the Data Object is available from
+// the provided URLs.
+message CreateDataObjectRequest {
+  // REQUIRED
+  // The data object to be created. The ID scheme is left up to the
+  // implementor but should be unique to the server instance.
+  DataObject data_object = 1;
 }
 
-message AddDataBundleObjectResponse {
-  string id = 1;
+message CreateDataObjectResponse {
+  // The ID of the created Data Object.
+  string data_object_id = 1;
 }
 
-message GetDataBundleObjectRequest {
-  string id = 1;
+message GetDataObjectRequest {
+  // REQUIRED
+  // The unique identifier for the requested Data Object.
+  string data_object_id = 1;
+
+  // OPTIONAL
+  // If provided will return the requested version of the selected Data Object.
+  string version = 2;
 }
 
-message GetDataBundleObjectChecksumRequest {
-  string checksum = 1;
+message GetDataObjectResponse {
+  // REQUIRED
+  // The Data Object that coincides with a specific GetDataObjectRequest.
+  DataObject data_object = 1;
 }
 
-message DeleteDataBundleObjectRequest {
-  string id = 1;
+message GetDataObjectVersionsRequest {
+  // REQUIRED
+  // The identifier of the Data Object you would like to retrieve all
+  // versions for.
+  string data_object_id = 1;
 }
 
-message DeleteDataObjectRequest {
-  string id = 1;
+message GetDataObjectVersionsResponse {
+  // REQUIRED
+  // All versions of the Data Objects that match the GetDataObjectVersions
+  // request.
+  repeated DataObject data_objects = 1;
 }
 
-message DataBundleStatusResponse {
+message UpdateDataObjectRequest {
```

Comment:
I think this might be out of sync with the underlying DataObject.
Propose that UpdateDataObject signature be:
`DataObject  updated = UpdateDataObject(DataObject update)`